solidity/ecdsa/test/Governable.test.ts

```diff
@@ -0,0 +1,177 @@
+              .connect(deployer)
+              .transferGovernance(ethers.Wallet.createRandom().address)
+          ).to.be.revertedWith("Caller is not the governance")
+        })
+      })
+
+      describe("when called by the governance", () => {
+        const newGovernance: string = ethers.Wallet.createRandom().address
+        let tx: ContractTransaction
+
+        before(async () => {
+          await createSnapshot()
+
+          tx = await governable
+            .connect(governance)
+            .transferGovernance(newGovernance)
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("updates governance address", async () => {
+          expect(await governable.governance()).to.be.equal(newGovernance)
+        })
+
+        it("emits GovernanceTransferred event", async () => {
+          await expect(tx)
+            .to.emit(governable, "GovernanceTransferred")
+            .withArgs(governance.address, newGovernance)
+        })
+      })
+
+      describe("when called by a third party", () => {
+        it("reverts", async () => {
+          await expect(
+            governable
+              .connect(thirdParty)
+              .transferGovernance(ethers.Wallet.createRandom().address)
+          ).to.be.revertedWith("Caller is not the governance")
+        })
+      })
+
+      describe("when new governance is zero address", () => {
+        it("reverts", async () => {
+          await expect(
+            governable
+              .connect(governance)
+              .transferGovernance(ethers.constants.AddressZero)
+          ).to.be.revertedWith("New governance is the zero address")
+        })
+      })
+    })
+  })
+
+  describe("_transferGovernance", () => {
+    beforeEach(async () => {
+      await createSnapshot()
+    })
+
+    afterEach(async () => {
+      await restoreSnapshot()
+    })
+
+    describe("when called by the deployer", () => {
+      it("succeeds", async () => {
+        const newGovernance = ethers.Wallet.createRandom().address
+
+        await governable
+          .connect(deployer)
+          ._transferGovernanceExposed(newGovernance)
+
+        expect(await governable.governance()).to.be.equal(newGovernance)
+      })
+    })
+
+    describe("when called by the governance", () => {
+      it("succeeds", async () => {
+        const newGovernance = ethers.Wallet.createRandom().address
+
+        await governable
+          .connect(governance)
+          ._transferGovernanceExposed(newGovernance)
+
+        expect(await governable.governance()).to.be.equal(newGovernance)
+      })
+    })
+
+    describe("when called by a third party", () => {
+      it("succeeds", async () => {
+        const newGovernance = ethers.Wallet.createRandom().address
+
+        await governable
+          .connect(thirdParty)
+          ._transferGovernanceExposed(newGovernance)
+
+        expect(await governable.governance()).to.be.equal(newGovernance)
+      })
```

Comment:
I think we should delete these two cases. `_transferGovernance` can be called only by the contract itself. If we want to test that it works, let's leave the deployer case because that's what is happening in the constructor case. Or even better, remove "when called by..." altogether because only the contract can call it.